doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1522,144 +1507,122 @@ Managing the local context
-   :name: pose proof
+   .. exn:: Proof is not complete.
+      :name: Proof is not complete. (assert)
 
-   This tactic behaves like :n:`assert @type {? as @simple_intropattern} by exact @term`
-   where :token:`type` is the type of :token:`term`. In particular,
-   :n:`pose proof @term as @ident` behaves as :n:`assert (@ident := @term)`
-   and :n:`pose proof @term as @simple_intropattern` is the same as applying the
-   :token:`simple_intropattern` to :token:`term`.
+      :n:`@ltac_expr3` was not able to prove the provided :n:`@term` or :n:`@one_term`.
 
-.. tacv:: epose proof @term {? as @simple_intropattern}
-   :name: epose proof
+   .. tacn:: eassert ( @ident : @type ) {? by @ltac_expr3 }
+             eassert @one_term {? @as_ipat } {? by @ltac_expr3 }
+      :name: eassert; _
 
-   While :tacn:`pose proof` expects that no existential variables are generated by
-   the tactic, :tacn:`epose proof` removes this constraint.
+      .. todo PR Need a better spot to define and comment on holes.
+         Maybe that could be in the existential variables chapter,
+         but making clear that they are existential variables?
 
-.. tacv:: pose proof (@ident := @term)
+      Unlike :tacn:`assert`, the :n:`@type`\s and :n:`@one_term`\s in
+      :tacn:`eassert` may contain :gdef:`holes <hole>`, denoted by :n:`_`,
+      for which the tactic will create existential variables.  This lets you
+      avoid specifying the asserted statement completely before starting to
+      prove it.
 
-   This is an alternative syntax for :n:`assert (@ident := @term)` and
-   :n:`pose proof @term as @ident`, following the model of :n:`pose
-   (@ident := @term)` but dropping the value of :token:`ident`.
+.. tacn:: enough ( @ident : @type ) {? by @ltac_expr3 }
+          enough @one_term {? @as_ipat } {? by @ltac_expr3 }
+   :name: enough; _
 
-.. tacv:: epose proof (@ident := @term)
+   .. todo PR The only difference vs assert is where the goal is added?  If so, then
+      this entry should only refer to assert and mention the difference(s).
+      There are several similar tactics in this section.  Maybe we should add a
+      summary of their differences?  Perhaps you could briefly outline them?
 
-   This is an alternative syntax for :n:`eassert (@ident := @term)`
-   and :n:`epose proof @term as @ident`, following the model of
-   :n:`epose (@ident := @term)` but dropping the value of
-   :token:`ident`.
+   The first form adds a new hypothesis :n:`@ident : @type` to the context
+   and adds :n:`@type` as a new subgoal after the current goal (rather than
+   before it as :tacn:`assert` would).  If present, :n:`@ltac_expr3` must solve the
+   initial goal entirely after the extra assumption :n:`@type` is added and possibly
+   destructed; otherwise, the tactic will fail.
 
-.. tacv:: enough (@ident : @type)
-   :name: enough
+   .. todo PR: must succeed for all subgoals and if no, the tactic fails?
+      and applies to assert, too?
```

Comment:
Yes. For `assert`, `ltac_expr3` is always applied only on one subgoal (which it must solve completely).